### PR TITLE
fix: tidy up Agentic Chat and Context accordions and behaviour

### PR DESCRIPTION
**Pull request by taiyab**
This PR merges the multiple accordions we currently have into one UX.

## Before
https://github.com/user-attachments/assets/0c11462a-9168-4882-9099-86a9a0a5843b

## After
https://github.com/user-attachments/assets/ac2ea91a-0443-487c-a55c-be4fc4a832c3

| Agentic Chat Selected | Agentic Chat Not Selected |
|----------------------|---------------------------|
| Agent Accordion | Standard Accordion |

## Test plan

This was tested locally, manually, in the following ways (where possible):

### Accordion Behavior Tests
- Verify single accordion expands/collapses correctly
- Confirm only one accordion section can be open at a time
- Test that accordion state persists between chat sessions
- Validate that accordion transitions are smooth without visual glitches

### Context Panel Tests
- Verify context information displays correctly when expanded
- Check that context updates properly when switching between different files/references
- Confirm context panel maintains scroll position when toggling
- Test context panel resize behavior

### Agentic Chat Integration
- Verify chat interface remains functional with new accordion structure
- Test chat history visibility when expanding/collapsing accordions
- Confirm chat input remains accessible regardless of accordion state
- Validate that chat notifications/indicators work with new layout

### Cross-browser Testing
- Test across multiple themes in VSCode and Jetbrains
- Verify responsive behavior on different window sizes

### Edge Cases
- Test with very long context content
- Verify behavior with multiple rapid accordion toggles
- Check keyboard navigation and accessibility
- Test with empty context/chat states

### CODE REVIEW

File: vscode/webviews/chat/cells/agenticCell/AgenticContextCell.tsx [hunk 1]
```diff
@@ -19,81 +29,181 @@ export const AgenticContextCell: FunctionComponent<{
...
+        processes,
+        intent,
+        onSwitchIntent,
+        contextItems,
+        isForFirstMessage,
+        model,
+        experimentalOneBoxEnabled,
+    }) => {
+        const [accordionValue, setAccordionValue] = useState<string | undefined>(() => {
+            return localStorage.getItem('agenticContextCell.accordionValue') || undefined
```
Comment (camdencheek) [inline vscode/webviews/chat/cells/agenticCell/AgenticContextCell.tsx:52]: Would recommend using `useLocalStorage` for this

Comment (camdencheek) [reply vscode/webviews/chat/cells/agenticCell/AgenticContextCell.tsx:52]: Oh, I see now that this is just copied from the previous code here. Feel free to ignore.

File: vscode/webviews/chat/cells/contextCell/ContextList.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            )
+        }
+        return (
+            <div className="tw-flex tw-flex-col tw-gap-2 tw-my-2 tw-border-t tw-border-t-muted tw-pt-4">
+                <div className="tw-text-sm tw-font-medium tw-text-foreground tw-flex tw-items-center tw-gap-3">
+                    <DatabaseIcon size={14} strokeWidth={1.75} className="tw-text-muted-foreground" />
+                    Context used
+                </div>
+                <ul className="tw-list-none tw-flex tw-flex-col tw-gap-1">
+                    {contextItems?.map((item, i) => (
```
Comment (camdencheek) [inline vscode/webviews/chat/cells/contextCell/ContextList.tsx:60]: This looks fully duplicated a few lines below. Can we clean that up?

File: vscode/webviews/components/codeSnippet/components/CodeExcerpt.module.css [hunk 3]
```diff
@@ -14,18 +14,25 @@
     :global(.line) {
         min-width: 1.5rem;
         text-align: right;
+        -webkit-user-select: none;
```
Comment (camdencheek) [inline vscode/webviews/components/codeSnippet/components/CodeExcerpt.module.css:17]: Do we actually need this? I think our CSS build tooling should add this if needed.

**Comment by taiyab** [comment]
@abeatrix Can I get a review?

File: vscode/webviews/chat/cells/agenticCell/AgenticContextCell.tsx [hunk 4]
```diff
@@ -142,3 +268,19 @@ const ProcessItem: FC<{
         </div>
     )
 }
+const getStateStyles = (state: string) => {
```
[the remark this answers is not in the document]
Comment (taiyab) [reply vscode/webviews/chat/cells/agenticCell/AgenticContextCell.tsx]: Eliminated this whole block and folded it into another.

File: vscode/webviews/chat/cells/messageCell/assistant/SearchResults.tsx [hunk 5]
```diff
@@ -343,28 +342,83 @@ export const SearchResults = ({
...
+                                                        selectedForContext={selectedFollowUpResults.has(
+                                                            result
+                                                        )}
+                                                        onSelectForContext={
+                                                            enableContextSelection
+                                                                ? handleSelectForContext
+                                                                : undefined
+                                                        }
+                                                    />
+                                                )}
```
[the remark this answers is not in the document]
Comment (taiyab) [reply vscode/webviews/chat/cells/messageCell/assistant/SearchResults.tsx]: Reformatted.

File: vscode/webviews/chat/cells/contextCell/ContextList.tsx (hunk 2 above)
Comment (camdencheek) [reply vscode/webviews/chat/cells/contextCell/ContextList.tsx:58]: This doesn't look resolved to me

File: vscode/webviews/chat/cells/messageCell/assistant/SearchResults.tsx [hunk 6]
```diff
@@ -339,30 +333,121 @@ export const SearchResults = ({
...
-                                                enableContextSelection
-                                                    ? handleSelectForContext
-                                                    : undefined
+                            <>
+                                {(() => {
+                                    const { boostedResults, nonBoostedResults } = useMemo(() => {
+                                        if (!boostedRepo) {
+                                            return {
+                                                boostedResults: resultsToShow,
+                                                nonBoostedResults: [],
```
Comment (camdencheek) [inline vscode/webviews/chat/cells/messageCell/assistant/SearchResults.tsx:342]: Can we move this logic out of the templating language? It's quite surprising to me to see variable definitions + function calls inside the template language, and I think it's actually a bug to call react hooks (like `useMemo`) conditionally.

**Comment by camdencheek** [comment]
I'm seeing some inconsistent rendering locally that might be related to these changes. We seemingly sometimes fail to render a separator line.

CleanShot 2025-02-03 at 11 12 09@2x

**Comment by camdencheek** [comment]
Can we make it a bit more obvious that the "context used" section is a sub-section? My initial reaction when testing this locally was to try to collapse the context used only to find it not collapsible and struggle to find that I need to click "Agentic chat" to collapse the (now very large) header

CleanShot 2025-02-03 at 11 29 46@2x

**Comment by camdencheek** [comment]
This looks really nice! The animations are ✨ 

I'll leave it to @abeatrix to hit approve since she's probably more familiar with the agentic chat domain

**Review by abeatrix — COMMENTED** [review]
Started the build locally and Agentic Chat seems to work as expected, really like the animation on toggle:

image

The font color for the failed command makes it a bit hard to read though:

image

Do we need to update the storybook for the updated components? Specifically for AgenticContextCell as the ContextList now lives in it.

Also it's a bit hard to test the PR because it involves changes beyond the Agentic Chat 
 and context according (e.g. RepositorySelector, SearchResults, Welcome Footer etc). Is it possible to split them into separated PRs so that it'd be easier to test and review? 🙇‍♀️
